awsEndpointGenerator/README.md

```diff
@@ -0,0 +1,106 @@
+
+# **AWS Endpoint Generator**
+
+## **Overview**
+The **AWS Endpoint Generator** is a Go-based tool that automatically fetches AWS service endpoint data from the **AWS SDK for Go v2** and generates Go files containing:
+
+1. **AWS-supported service endpoints** across partitions and regions.
+2. **Service ID constants** to facilitate programmatic access to service metadata.
+
+This tool helps in determining **AWS service availability** across different regions and partitions.
+
+---
+
+## **How It Works**
+The tool retrieves endpoint data from the following AWS SDK JSON file:  
+🔗 **[AWS Endpoints JSON](https://raw.githubusercontent.com/aws/aws-sdk-go-v2/master/codegen/smithy-aws-go-codegen/src/main/resources/software/amazon/smithy/aws/go/codegen/endpoints.json)**  
+
+It then **parses the JSON**, extracts relevant data, and generates two Go files:
+- **`awsSupportedEndpoints/endpoints_gen.go`** → Contains partition, region, and service endpoint definitions.
+- **`awsSupportedEndpoints/service_id_gen.go`** → Contains service ID constants for all AWS services.
+
+---
+
+## **Installation**
+### **1. Move to the dir steampipe-plugin-aws**
+```sh
+cd steampipe-plugin-aws
+```
+
+#### **2. Updated `main.go` Content with:**
```

Comment:
Can we move this out of the main package and instead have it be a nested package that can be run on its own?